ansi2html/converter.py

```diff
@@ -786,6 +787,8 @@ def main() -> None:
         title=opts.output_title,
     )
 
+    reader = io.TextIOWrapper(sys.stdin.buffer, opts.input_encoding, "replace")
```

Comment:
Can you remind me (while lost) why we need this wrapper in the first place?

Reply:
As far as I know, that's the Python 3 way to making `sys.stdin` read anything with a different encoding than utf-8.

Reply:
I see, argument `--input-encoding=ENCODING`! And there is no `.detach()` now because you're going straight to the bytes buffer. I think there are two options here: (a) Only wrap if ther is a bytes layer below (a la old approach) or (b) require binary buffer and have the tests make it work, always wrap.

I must say that the difference in noise in the test suite for (a) with `StringIO` is significantly lower, so I'd personally be in favor of the old approach:
https://github.com/pycontribs/ansi2html/blob/751554082d493d5e122f1437789d77a10314a458/ansi2html/converter.py#L721-L725

The commend could say something like "if there is no binary .buffer beneath it, e.g. with StringIO" if you'd like more detail there as well, just an idea.

Reply:
I switched from `sys.stdin.detach()` to `sys.stdin.buffer`, because it originally failed with `"TextIO" has no attribute "detach"`.

The old approach with catching an exception in production code just to make the tests work does smell a bit. And also 
```python
try:

except Error: 
    pass
```
is an anti-pattern that should not have been used in the first place.